### DISCUSSION TITLE
Water Mask

### DISCUSSION THREAD

**Discussion by Rick0219** [alexisInSAR/EZ-InSAR#38] [Help]
The study area I have chosen has a lot of water and I want to know will ez_insar mask the water area when unwrapping with isce or should I do this step by myself?

**Comment by alexisInSAR** [comment]
Hi, 

This is an ongoing discussion between the developers as the current version is not able to provide a robust solution for water bodies. 

However several options are possible: 
1. Create a SLC stack and use StaMPS to estimate the ground surface displacements. In this case, the water bodies will be removed. 
2. If you want to use MintPy, you can set a water mask in the parameters list. Unfortunately, the unwrapping will be done on these areas. In this case, it is possible to use a DEM for which the water bodies have no-value data. If I remember correctly, the interferograms will have no value and the unwrapping should not be performed on these pixels. 

I know that ISCE has the possibility to mask the water bodies, but this option is not fully implemented in EZ-InSAR (probability in a next version). 

Kind regards, 

Alexis

**Reply by Rick0219** [reply]
Sorry for my late reply,I wonder know which dem product will have no value on water bodies.Or I should set the value NAN by myself?

**Reply by alexisInSAR** [reply]
For the DEM, it depends on the types, sources, repository, etc. 

You can just download the SRTM and manually replace the water bodies by NaN.

**Reply by forgetsyl** [reply]
Dear alexisInSAR, as you suggested, I changed the dem to a nan value and then calculated the unw. However, the results are not very good, and the calculation time seems to be the same as it was before I changed the dem.
